src/test/java/rx/plugins/RxJavaHooksTest.java

```diff
@@ -0,0 +1,165 @@
+            
+            Throwable ex = ts.getOnErrorEvents().get(0);
+
+            Assert.assertTrue("" + ex, ex instanceof AssemblyStackTraceException);
+
+            Assert.assertTrue("" + ex.getCause(), ex.getCause() instanceof TestException);
+
+            Assert.assertTrue(ex.getMessage(), ex.getMessage().contains("createSingle"));
+        } finally {
+            RxJavaHooks.resetAssemblyTracking();
+        }
+    }
+    
+    static Completable createCompletable() {
+        return Completable.error(new Func0<Throwable>() {
+            @Override
+            public Throwable call() {
+                return new TestException();
+            }
+        });
+    }
+    
+    @Test
+    public void assemblyTrackingCompletable() {
+        RxJavaHooks.enableAssemblyTracking();
+        try {
+            TestSubscriber<Integer> ts = TestSubscriber.create();
+            
+            createCompletable().subscribe(ts);
+            
+            ts.assertError(AssemblyStackTraceException.class);
+            
+            Throwable ex = ts.getOnErrorEvents().get(0);
+
+            Assert.assertTrue("" + ex, ex instanceof AssemblyStackTraceException);
+
+            Assert.assertTrue("" + ex.getCause(), ex.getCause() instanceof TestException);
+
+            Assert.assertTrue(ex.getMessage(), ex.getMessage().contains("createCompletable"));
+        } finally {
+            RxJavaHooks.resetAssemblyTracking();
+        }
+    }
+    
+    @SuppressWarnings("rawtypes")
+    @Test
+    public void lockdown() throws Exception {
+        RxJavaHooks.reset();
+        RxJavaHooks.lockdown();
+        try {
+            Action1 a1 = Actions.empty();
+            Func1 f1 = UtilityFunctions.identity();
+            Func2 f2 = new Func2() {
+                @Override
+                public Object call(Object t1, Object t2) {
+                    return t2;
+                }
+            };
+            
+            for (Method m : RxJavaHooks.class.getMethods()) {
+                if (m.getName().startsWith("setOn")) {
+                    
+                    Method getter = RxJavaHooks.class.getMethod("get" + m.getName().substring(3));
+                    
+                    Object before = getter.invoke(null);
+                    
+                    if (m.getParameterTypes()[0].isAssignableFrom(Func1.class)) {
+                        m.invoke(null, f1);
+                    } else
+                    if (m.getParameterTypes()[0].isAssignableFrom(Action1.class)) {
+                        m.invoke(null, a1);
+                    } else {
+                        m.invoke(null, f2);
+                    }
+                    
+                    Object after = getter.invoke(null);
+                    
+                    Assert.assertSame(m.toString(), before, after);
+                }
+            }
+            
+        } finally {
+            RxJavaHooks.lockdown = false;
```

Comment:
It seems like this should have a method for setting in tests, package-local is a little annoying unless this is intended only for RxJava tests. Is it?